core/jobs/batch_jobs/audit_exploration_jobs.py

```diff
@@ -0,0 +1,1479 @@
+          List[exp_models.ExplorationModel],
+          List[opportunity_models.ExplorationOpportunitySummaryModel]
+    ]) -> Tuple[
+        Optional[exp_models.ExplorationModel],
+        Optional[opportunity_models.ExplorationOpportunitySummaryModel]
+    ]:
+        """Returns the pair of exp and opportunity models.
+
+        Args:
+            models_list_pair: tuple. The pair of models list.
+
+        Returns:
+            tuple. The pair of exp and opportunity models.
+        """
+        exp_model = None
+        opportunity_model = None
+        if len(models_list_pair[0]) == 1:
+            exp_model = models_list_pair[0][0]
+        if len(models_list_pair[1]) == 1:
+            opportunity_model = models_list_pair[1][0]
+        model_pair = (exp_model, opportunity_model)
+        return model_pair
+
+    @staticmethod
+    def filter_invalid_ref_exp_id(
+        states_dict: Dict[str, state_domain.State]
+    ) -> List[Dict[str, List[Dict[str, List[str]]]]]:
+        """Returns the errored state values, validates
+            - refresher_exploration_id should be None for all lessons
+
+        Args:
+            states_dict: dict[str, State]. The dictionary containing
+                state name as key and State object as value.
+
+        Returns:
+            states_with_values: list[dict]. The list of dictionaries
+            containing the errored values.
+        """
+        states_with_values = []
+
+        for state_name, state in states_dict.items():
+            invalid_refresher_exploration_id = []
+            answer_groups = state.interaction.answer_groups
+
+            for answer_group in answer_groups:
+                answer_group_index = str(answer_groups.index(answer_group))
+
+                # Validates refresher_exploration_id be None for all lessons.
+                if answer_group.outcome.refresher_exploration_id is not None:
+                    invalid_refresher_exploration_id.append(
+                        f'The refresher_exploration_id '
+                        f'of answer group {answer_group_index} '
+                        f'is not None having value '
+                        f'{answer_group.outcome.refresher_exploration_id}'
+                    )
+
+            default_outcome = state.interaction.default_outcome
+            if default_outcome is not None:
+                if default_outcome.refresher_exploration_id is not None:
+                    invalid_refresher_exploration_id.append(
+                        f'The refresher_exploration_id '
+                        f'of default outcome '
+                        f'is not None having value '
+                        f'{default_outcome.refresher_exploration_id}'
+                    )
+
+            states_with_values.append(
+                {'state_name': state_name,
+                'invalid_refresher_exploration_id': (
+                    invalid_refresher_exploration_id)
+                }
+            )
```

Comment:
Should you wrap this in an `if` statement? Of do you have a check elsewhere to make sure you aren't printing out empty errors to the Beam Job output.
```suggestion
        if invalid_refresher_exploration_id:
            states_with_values.append(
                {'state_name': state_name,
                'invalid_refresher_exploration_id': (
                    invalid_refresher_exploration_id)
                }
            )
```

Reply:
I just saw that you have a filtering function already,